src/sql/src/statements/show.rs

```diff
@@ -53,19 +53,15 @@ pub struct ShowColumns {
 
 impl Display for ShowColumns {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str("SHOW")?;
+        write!(f, "SHOW ")?;
         if self.full {
-            f.write_str(" FULL")?;
+            write!(f, "FULL ")?;
         }
-
-        let table = format!("IN {}", &self.table);
-        let database = match &self.database {
-            Some(database) => format!("IN {}", database),
-            None => String::default(),
-        };
-
-        let kind = &self.kind;
-        write!(f, r#" COLUMNS {table} {database} {kind}"#)
```

Comment:
@irenjj I pushed a commit to showcase my suggestions here.

In the previous pattern, ` {database} ` can still have an extra whitespace if database is empty.

And why I'd prefer this fallthrough way to the construct and then fill the string template?

This is because formatting (display a struct) is a linear job. If you write code like:

```rust
        let if_not_exists = self.format_if_not_exists();
        let name = &self.name;
        let columns = format_list_indent!(self.columns);
        let constraints = self.format_constraints();
        let partitions = self.format_partitions();
        let engine = &self.engine;
        let options = self.format_options();
        let maybe_external = if self.engine == FILE_ENGINE {
            "EXTERNAL "
        } else {
            ""
        };
        write!(
            f,
            r#"CREATE {maybe_external}TABLE {if_not_exists} {name} (
{columns},
{constraints}
)
{partitions}ENGINE={engine}
{options}"#
        )
```

You may jump up and down to check the template is correct, and every part is correct. But we're just printing chars, so we can `write!` on the fly.

However, I noticed that the original implementation you may refer to https://github.com/GreptimeTeam/greptimedb/pull/1336 by @killme2008. So we still have some existing code to migrate. Then I'm OK as long as the result is correct (no extra whitespace) and we gradually move to the flavor I proposed here:

```rust
impl Display for ShowTables {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "SHOW ")?;
        if self.full {
            write!(f, "FULL ")?;
        }
        write!(f, "TABLES ")?;
        if let Some(database) = &self.database {
            write!(f, "IN {database} ")?;
        }
        write!(f, "{}", &self.kind)
    }
}
```

Reply:
Seems only

* CreateExternalTable
* CreateTable

doesn't follow this pattern. But we can leave it to a follow up. This patch is already too large.